statviz/src/views/Statistics/components/TopProductsPieChart.tsx

```diff
@@ -0,0 +1,82 @@
+import PieChart from "../../../components/nivo-graphs/PieChart";
+import { Sort, table } from "../../../utils/table";
+import {
+  CreatedBoxesData,
+  CreatedBoxesResult,
+  ProductDimensionInfo,
+  QueryCreatedBoxesArgs,
+} from "../../../types/generated/graphql";
+import { ApolloError, useQuery, gql } from "@apollo/client";
+import { Heading } from "@chakra-ui/react";
+import { round } from "lodash";
+
+const CREATED_BOXES_QUERY = gql`
+  query createdBoxes($baseId: Int!) {
+    createdBoxes(baseId: $baseId) {
+      facts {
+        boxesCount
+        productId
+        categoryId
+        createdOn
+        gender
+        itemsCount
+      }
+      dimensions {
+        product {
+          id
+          name
+        }
+        category {
+          id
+          name
+        }
+      }
+    }
+  }
+`;
+
+export default function TopProductsPieChart() {
+  const { data, loading, error } = useQuery<
+    CreatedBoxesData,
+    QueryCreatedBoxesArgs
+  >(CREATED_BOXES_QUERY, { variables: { baseId: 1 } });
+
+  if (error instanceof ApolloError) {
+    return <p>{error.message}</p>;
+  }
+  if (loading) {
+    return <p>loading...</p>;
+  }
+
+  let createdBoxes = table(data.createdBoxes.facts as CreatedBoxesResult[]);
```

Comment:
please memoize and try to not use let
https://eslint.org/docs/latest/rules/prefer-const